lib/reporters/teamcity_reporter.js

```diff
@@ -38,7 +38,9 @@ TeamcityReporter.prototype = {
     if (result.skipped) {
       this.out.write('##teamcity[testIgnored name=\'' + this._namify(prefix, result) + '\' message=\'pending\']\n');
     } else if (!result.passed) {
-      this.out.write('##teamcity[testFailed name=\'' + this._namify(prefix, result) + '\' message=\'' + escape(result.error.message) + '\' details=\'' + escape(result.error.stack) + '\']\n');
+      var message = (result.error && result.error.message) || '';
+      var stack = (result.error && result.error.stack) || '';
+      this.out.write('##teamcity[testFailed name=\'' + this._namify(prefix, result) + '\' message=\'' + escape(message) + '\' details=\'' + escape(stack) + '\']\n');
```

Comment:
It doesn't really make any difference for teamcity. It uses those fields ```message``` & ```details``` to provide extra information when inspecting a failed test. 

For example: 
* The first test on the image below shows what happens when those fields are empty.
* On the second test they are not empty. 
![screen shot 2016-08-31 at 3 06 40 pm](https://cloud.githubusercontent.com/assets/118375/18142944/d3624b30-6f8d-11e6-9e62-d236a38badbc.png)

Before this PR you don't get any details because the process exits with an error.
![screen shot 2016-08-31 at 3 13 02 pm](https://cloud.githubusercontent.com/assets/118375/18143018/2e3e862c-6f8e-11e6-8fd9-6448996f6938.png)